lib/datapacksutils.js

```diff
@@ -3042,8 +3042,135 @@ DataPacksUtils.prototype.loadSalesforceMetadata = async function() {
+            }
+        }
+    });
+
+    VlocityUtils.report('Retrieving Salesforce Metadata Dependencies');
+
+    let dependenciesArr = [];
+    for (let key of Object.keys(selectedMetadataMappings)) {
+        try {
+            let query = `Select MetadataComponentName, MetadataComponentType, RefMetadataComponentName, RefMetadataComponentType from MetadataComponentDependency WHERE RefMetadataComponentType = '${key}'`;
+            VlocityUtils.report('Query', query);
+            let foundMetadataComponents = await this.vlocity.jsForceConnection.tooling.query(query);
+            VlocityUtils.report('Found Records', foundMetadataComponents.records.length);
+            dependenciesArr.push(foundMetadataComponents);
+        } catch (err) {
+            VlocityUtils.error('Query Error', err);
+        }
+    }
+
+    dependenciesArr = dependenciesArr.concat(await this.retrieveExceptionalMetadataDependencies(jobInfo, salesforceTypes));
+    
+    let allComponentMetadataDependencies = [];
+    dependenciesArr.forEach(item => {
+        allComponentMetadataDependencies = allComponentMetadataDependencies.concat(item.records);
+    });
+
+    let metadataComponentMappings = {};
+    allComponentMetadataDependencies.forEach(item => {
+        let keyPrefix = allMetaDataPrefixMappings[item.RefMetadataComponentType];
+        let keySuffix = allMetaDataSuffixMappings[item.RefMetadataComponentType];
+
+        let valuePrefix = allMetaDataPrefixMappings[item.MetadataComponentType];
+        let valueSuffix = allMetaDataSuffixMappings[item.MetadataComponentType];
+
+        if (keyPrefix != undefined && keySuffix != undefined && valuePrefix != undefined && valueSuffix != undefined) {
+            let key = keyPrefix + '/' + item.RefMetadataComponentName + (keySuffix ? '.' + keySuffix : '');
+            let value = valuePrefix + '/' + item.MetadataComponentName + (valueSuffix ? '.' + valueSuffix : '');
+            
+            if (!metadataComponentMappings[key]) {
+                metadataComponentMappings[key] = [];
+            }
+            metadataComponentMappings[key].push(value);
+        }
+    });
+
+    fs.outputFileSync(path.join(sfFolderPath, 'SalesforceReferences.yaml'), yaml.safeDump(metadataComponentMappings), "UTF-8");
+
+    let newSFKeysToBeAddedToManifest = [];
+    salesforceKeys.forEach(item => {
+        if (metadataComponentMappings[item]) {
+            metadataComponentMappings[item].forEach(component => {
+                if (!salesforceKeys.includes(component)) {
+                    newSFKeysToBeAddedToManifest.push(component);
+                }
+            })
+        }
+    });
+
+    return newSFKeysToBeAddedToManifest;
+}
+
+DataPacksUtils.prototype.retrieveExceptionalMetadataDependencies = async function(jobInfo, salesforceTypes) {
+    var exceptionDetailsObj = {},
+        dependenciesArr = [],
+        exceptionsArr = ['ApexClass'];
```

Comment:
You can just hardcode the instantiation of the exceptionsArr. The rest of the function is perfect for adding a second type without having to re-code it all. But this first part can still be "hardcoded"